openedx/core/djangoapps/waffle_utils/__init__.py

```diff
@@ -3,8 +3,35 @@
 
 Includes namespacing, caching, and course overrides for waffle flags.
 
-For testing WaffleFlags, see testutils.py.
-For testing WaffleSwitchNamespace, use its context manager.
+Usage:
+
+For Waffle Flags, first set up the namespace, and then create flags using the
+namespace.  For example:
+
+    WAFFLE_FLAG_NAMESPACE = WaffleFlagNamespace(namespace='course_experience', log_prefix=u'Course Experience: ')
```

Comment:
I decided to go with `name`.  The code I imagined would be less clear didn't really exist.

I agree the `log_prefix` is weird, and I don't get why it is needed at all, since it is repeated in the flag/switch name in the log messages.  Given that, it's used for Grades, and I don't want to just delete it without @nasthagiri stating it is ok.  Even if the log statements need to start with the "label" for some reason, I don't see why they couldn't just start with the name(space) and leave it at that.